VRCFaceTracking.Core/Models/OscTarget.cs

```diff
@@ -9,18 +10,94 @@ public partial class OscTarget : ObservableObject, IOscTarget
     [ObservableProperty] private bool _isConnected;
 
     [ObservableProperty] [property: SavedSetting("OSCInPort", 9001)]
+
     private int _inPort;
 
     [ObservableProperty] [property: SavedSetting("OSCOutPort", 9000)]
+
     private int _outPort;
 
-    [ObservableProperty] 
+    [ObservableProperty]
     [NotifyPropertyChangedFor(nameof(InPort))] [NotifyPropertyChangedFor(nameof(OutPort))]
+
     [property: SavedSetting("OSCAddress", "127.0.0.1")]
     private string _destinationAddress;
-    
+
+    private const string DefaultAddress = "127.0.0.1";
+    private const int DefaultInPort = 9001;
+    private const int DefaultOutPort = 9000;
+    private bool _isReverting = false;
+
     public OscTarget(ILocalSettingsService localSettingsService)
     {
         PropertyChanged += (_, _) => localSettingsService.Save(this);
     }
-}
\ No newline at end of file
+
+    /// Called after the DestinationAddress property has changed.
+    /// It validates the new address and reverts it to the default if invalid.
+    partial void OnDestinationAddressChanged(string oldValue, string newValue)
+    {
+        // Prevent re-entrancy if we're already reverting
+        if (_isReverting)
+        {
+            return;
+        }
+
+        _isReverting = true;
+
+        // Check if the new address is valid and revert if invalid
+        if (!IsValidAddress(newValue))
+        {
+            DestinationAddress = DefaultAddress;
+        }
+
+        _isReverting = false;
+    }
+
+    /// Called after the InPort property has changed.
+    /// It checks if the InPort is valid and reverts to the default if it's invalid (less than 1).
+    partial void OnInPortChanged(int oldValue, int newValue)
+    {
+        if (newValue < 1)
+        {
+            InPort = DefaultInPort;
+        }
+    }
+
+    /// Called after the OutPort property has changed.
+    /// It checks if the OutPort is valid and reverts to the default if it's invalid (less than 1).
+    partial void OnOutPortChanged(int oldValue, int newValue)
+    {
+        if (newValue < 1)
+        {
+            OutPort = DefaultOutPort;
+        }
+    }
```

Comment:
I worry this will cause an infinite loop. Also, such functionality could be achieved by editing values in the ui xaml here https://github.com/benaclejames/VRCFaceTracking/blob/3480ca415c08d0d1e7cdb2a4a8e298beca87663c/VRCFaceTracking/Views/SettingsPage.xaml#L64
and here
https://github.com/benaclejames/VRCFaceTracking/blob/3480ca415c08d0d1e7cdb2a4a8e298beca87663c/VRCFaceTracking/Views/SettingsPage.xaml#L78

Reply:
Also, not sure you want to be using these operators. They both make it so that the port can only be set if the port is **less than** 1..

Reply:
yeah i will use that instead than the port check does not have to exist,

As for the IP verification i might change the way i approach that since host names are disallowed as it will throw an exception in 
https://github.com/benaclejames/VRCFaceTracking/blob/25f7e9548505978be648e8ecf3a507afc14103c3/VRCFaceTracking.Core/Services/OscRecvService.cs#L49

It might also be smart to do this in 
https://github.com/benaclejames/VRCFaceTracking/blob/25f7e9548505978be648e8ecf3a507afc14103c3/VRCFaceTracking.Core/Services/OscRecvService.cs#L30C1-L51C6

As there is already some handling in there but it seems to actually not handle empty strings but does handle NULL ref

Reply:
Although this would still cause the application to crash on restart since it doesn't reset the value.